src/roiextractors/extractors/femtonicsimagingextractor/femtonicsimagingextractor.py

```diff
@@ -0,0 +1,343 @@
+        session_key = f"MSession_{self._munit}"
+        munit_key = f"MUnit_{self._munit}"
+        channel_key = f"Channel_{self._selected_channel_index}"
+
+        if (
+            session_key in self._file
+            and munit_key in self._file[session_key]
+            and channel_key in self._file[session_key][munit_key]
+        ):
+            self._video = DatasetView(self._file[session_key][munit_key][channel_key])
+        else:
+            raise Exception(f"Cannot find data at {session_key}/{munit_key}/{channel_key}")
+
+    def get_available_channels_from_file(self) -> List[str]:
+        """Get available channels from the current file."""
+        session_key = f"MSession_{self._munit}"
+        munit_key = f"MUnit_{self._munit}"
+
+        if session_key not in self._file or munit_key not in self._file[session_key]:
+            raise ValueError(f"MUnit {self._munit} not found in file")
+
+        attrs = dict(self._file[session_key][munit_key].attrs)
+        num_channels = attrs.get("VecChannelsSize", 2)
+
+        channels = []
+        for i in range(num_channels):
+            name_arr = attrs.get(f"Channel_{i}_Name", [])
+            name = self._decode_string(name_arr) if len(name_arr) > 0 else f"Channel_{i}"
+            channels.append(name)
+        return channels
+
+    def _decode_string(self, arr) -> str:
+        """Convert int16 array to string."""
+        if len(arr) == 0:
+            return ""
+        return "".join(chr(x) for x in arr if x != 0)
+
+    def get_image_shape(self) -> Tuple[int, int]:
+        """Get the shape of the video frame (num_rows, num_columns)."""
+        return self._video.shape[1], self._video.shape[2]  # (height, width)
+
+    def get_num_samples(self) -> int:
+        """Get the number of samples (frames) in the video."""
+        return self._video.shape[0]
+
+    def get_channel_names(self) -> List[str]:
+        """Get the channel names."""
+        return [self._selected_channel_name]
+
+    def get_sampling_frequency(self) -> float:
+        """Get the sampling frequency in Hz."""
+        time_per_frame_ms = self._get_time_per_frame()
+        if time_per_frame_ms is None:
+            raise ValueError("Sampling frequency could not be determined from metadata.")
+        return 1000.0 / time_per_frame_ms
+
+    def _get_time_per_frame(self) -> Optional[float]:
+        """Get time per frame from metadata."""
+        session_key = f"MSession_{self._munit}"
+        munit_key = f"MUnit_{self._munit}"
+        attrs = dict(self._file[session_key][munit_key].attrs)
+        return attrs.get(
+            "ZAxisConversionConversionLinearScale"
+        )  # ZAxisConversionConversionLinearScale gives the duration of one frame in milliseconds.
+
+    def get_dtype(self) -> np.dtype:
+        """Get the data type of the video."""
+        return self._video.dtype
+
+    def get_frames(self, frame_idxs: ArrayType, channel: Optional[int] = 0):
```

Comment:
We don't need this method. It is not really necessary and we aim to remove it at some point.